src/Makevars

```diff
@@ -14,7 +14,7 @@ $(STATLIB):
 	curdir=$(CURDIR)
 	cd $(RPOLARS_RUST_SOURCE)/..
 	cargo build --lib --release --manifest-path="$(RPOLARS_RUST_SOURCE)/Cargo.toml"
-	cd $(curdir)
+	cd "$$curdir"
```

Comment:
What is the reason for this change?

Reply:
Like I've mentioned in the beginning, `curdir=$(CURDIR)` on line 14 is a bash command assigning an environment variable with name `curdir`. `$(curdir)` will be evaluate to empty before build since `curdir` is not declared as a Makevars variable, and originally line 17 will become `cd` in build, and this leads to error in Nix derivation. `"$$curdir"` will  be evaluated to `"$curdir"` before build and this will be interpreted as the value in the environment variable in build.

Reply:
Thank you for the explanation. Sorry, I missed your comment.......

Reply:
I'm unsure how `CURDIR` is maintained by the build system. If it constantly points to the working directory then using it after `cd` may not be ideal. Otherwise if it is only assigned at the start we can directly use it as you suggest.

Reply:
I'm quite new to Make.

Reply:
@eitsupi @Sicheng-Pan  Do I understand this correctly? GNU docs / R makevars tradition favors $(CURDIR), however this might not always work on any thinkable system and there "$$curdur" e.g. flake. Maybe "$$curdir" will not work somewhere also, but we don't know that yet. Maybe we can print the acquired value of {curdir} to assist a future bug report.

Reply:
Sorry, I did not check this Makefile properly. I will submit another PR to update this.

Reply:
@sorhawell I'm simply not sure about how `CURDIR` is introduced and maintained by the build system, so I'm not sure if it will remain the same after changing current working directory with `cd` . But at least it is correct at the beginning so it might be reasonable to assign it to a environment variable `curdir`, and its value should be `./src` (the folder containing the Makevars file).